core/core.ts

```
@@ -286,11 +288,17 @@ export class Core {
         ]),
       ])({ ...provider });
 
-      for (const site of siteIndexingOptions) {
-        await this.getEmbeddingsProviderAndIndexDoc(site, msg.data.reIndex);
-      }
+      // Create an array of promises for each site's indexing process
+      const indexingPromises = siteIndexingOptions.map(site =>
+        this.getEmbeddingsProviderAndIndexDoc(site, msg.data.reIndex)
+      );
 
-      this.ide.infoPopup("Docs indexing completed");
+      // Await for all promises to resolve
+      try {
+        await Promise.all(indexingPromises);
+      } catch (error: any) {
+        console.error(error.message);
```

This should probably continue to throw rather than silently fail (from the user perspective).